ubuntu-advantage

```diff
@@ -313,10 +325,50 @@ is_livepatch_enabled() {
     canonical-livepatch status >/dev/null 2>&1
 }
 
-validate_user_pass_token(){
+validate_user_pass_token() {
     echo "$1" | grep -q '^[^:]\+:[^:]\+$'
 }
 
+check_url_with_credentials() {
+    local repo_url="$1"
+    local credentials="$2"
+    local file="$3"
+
+    local url
+    url=$(private_repo_url "$repo_url" "$credentials" "$file")
+
+    local tempfile log
+    tempfile="$(mktemp)"
+    log="$(mktemp)"
+    if "$APT_HELPER" download-file "$url" "$tempfile" >"$log" 2>&1; then
+        rm -f "$tempfile" "$log"
```

Comment:
We only need this tempfile because apt-helper won't write to /dev/null, right? Perhaps add a quick comment here about that, and that tempfile will just be discarded? Maybe even rename it to "discardfile" or something like that :)

Reply:
done